### PR TITLE
Rspec enumerable

### PR DESCRIPTION
**Pull request by abdulkhaliqdev**
## Rspec Test for Enumerable it have following things
- Test for my_each 
- Test for my_each_with_index
- Test for my_select
- Test for my_any?
- Test for my_none?
- Test for my_map
- Test for my_count
- Test for my_all?
- Test for my_inject

### CODE REVIEW

**Review by rindrajosia — CHANGES_REQUESTED** [review]
**Changes required** 

Hello team!!

Well done job so far. However, you have to implement test case for  `my_none?` methode to get your project approved and delete inline comment.
I am sure that it won't take much time.

Please submit for another review after making the required changes.
Happy coding !!!

File: lib/enumerable_method.rb [hunk 1]
```diff
@@ -70,18 +67,15 @@ def my_all?(parameter = nil)
   def my_any?(parameter = nil)
     # for parameter
```
Comment (oloomoses) [inline lib/enumerable_method.rb]: - [x] remove the inline comment

File: lib/enumerable_method.rb [hunk 2]
```diff
@@ -70,18 +67,15 @@ def my_all?(parameter = nil)
...
+      if parameter.instance_of? Class
+        my_each { |i| flag = true if parameter === i }
+      elsif parameter.instance_of? Regexp
+        my_each { |i| flag = true if parameter.match? i }
       else
-        false
+        my_each { |item| flag = true if parameter == item }
       end
+      flag
     # for black_given
```
Comment (oloomoses) [inline lib/enumerable_method.rb]: - [ ] remove the inline comment

File: lib/enumerable_method.rb [hunk 3]
```diff
@@ -102,26 +96,29 @@ def my_any?(parameter = nil)
   def my_none?(parameter = nil)
     # for parameter
```
Comment (oloomoses) [inline lib/enumerable_method.rb]: - [ ] remove the inline comment

File: lib/enumerable_method.rb [hunk 4]
```diff
@@ -166,7 +164,7 @@ def my_inject(pa1 = nil, pa2 = nil)
     data = to_a
     # Through Error if Parameter and block are not given
```
Comment (oloomoses) [inline lib/enumerable_method.rb]: - [ ] remove the inline comment

File: lib/enumerable_method.rb [hunk 5]
```diff
@@ -166,7 +164,7 @@ def my_inject(pa1 = nil, pa2 = nil)
...
     if pa1.nil? && pa2.nil? && !block_given?
-      return  'Required Block or Parameter'
+      raise LocalJumpError
     # Block is given and parameter is given but its value is nil
```
Comment (oloomoses) [inline lib/enumerable_method.rb]: - [ ] remove the inline comment

**Review by oloomoses — CHANGES_REQUESTED** [review]
# Status: Changes Required
### Hello team, you have worked so hard on this project, and it's almost ready. :muscle: 
### Kindly ensure that there are no inline comments in your code for it to be approved. This should not take you long.

**Review by oloomoses — APPROVED** [review]
# Status: Approved
### Congratulations team :muscle: :tada: :100: 
### You can now merge and close this Pull Request. I wish you the best on your next project.

**Pull request closed without merging**